net_sink.go

```diff
@@ -109,12 +109,10 @@ func NewNetSink(opts ...SinkOption) FlushableSink {
 		bufSize = defaultBufferSizeTCP
 	}
 
-	outc := make(chan *bytes.Buffer, approxMaxMemBytes/bufSize)
-	writer := &sinkWriter{
-		outc: outc,
-	}
-	s.outc = outc
+	s.outc = make(chan *bytes.Buffer, approxMaxMemBytes/bufSize)
+	s.retryc = make(chan *bytes.Buffer, approxMaxMemBytes/bufSize)
```

Comment:
nit: size should be 1?

```suggestion
	s.retryc = make(chan *bytes.Buffer, 1)
```

Reply:
I was originally thinking that we don't want to create backpressure here, instead having the bottleneck point be writing into the `outc`. But realizing that when we first write into this channel, we'll `continue` and we preferentially process the retry queue, I think we could reasonably make this 1. Will add.